internal/otelcollector/config/metricgateway/config_builder_test.go

```diff
@@ -282,21 +282,157 @@ func TestMakeConfig(t *testing.T) {
 					testutils.NewMetricPipelineBuilder().
 						WithName("test").
 						WithOTLPInput(false).
 						WithOTLPOutput(testutils.OTLPEndpoint("https://localhost")).Build(),
 				},
 			},
+			{
+				name:           "pipeline with runtime input and namespace include filter",
+				goldenFileName: "runtime-namespace-include.yaml",
+				pipelines: []telemetryv1alpha1.MetricPipeline{
+					testutils.NewMetricPipelineBuilder().
+						WithName("cls").
+						WithRuntimeInput(true, testutils.IncludeNamespaces("kyma-system", "default")).
+						WithOTLPOutput(testutils.OTLPEndpoint("https://localhost")).Build(),
+				},
+			},
+			{
+				name:           "pipeline with runtime input and namespace exclude filter",
+				goldenFileName: "runtime-namespace-exclude.yaml",
+				pipelines: []telemetryv1alpha1.MetricPipeline{
+					testutils.NewMetricPipelineBuilder().
+						WithName("cls").
+						WithRuntimeInput(true, testutils.ExcludeNamespaces("kube-system", "istio-system")).
+						WithOTLPOutput(testutils.OTLPEndpoint("https://localhost")).Build(),
+				},
+			},
+			{
+				name:           "pipeline with prometheus input and namespace filters",
+				goldenFileName: "prometheus-namespace-filters.yaml",
+				pipelines: []telemetryv1alpha1.MetricPipeline{
+					testutils.NewMetricPipelineBuilder().
+						WithName("cls").
+						WithPrometheusInput(true, testutils.IncludeNamespaces("monitoring", "observability")).
+						WithOTLPOutput(testutils.OTLPEndpoint("https://localhost")).Build(),
+				},
+			},
+			{
+				name:           "pipeline with istio input and namespace filters",
+				goldenFileName: "istio-namespace-filters.yaml",
+				pipelines: []telemetryv1alpha1.MetricPipeline{
+					testutils.NewMetricPipelineBuilder().
+						WithName("cls").
+						WithIstioInput(true, testutils.ExcludeNamespaces("kube-system")).
+						WithOTLPOutput(testutils.OTLPEndpoint("https://localhost")).Build(),
+				},
+			},
+			{
+				name:           "pipeline with OTLP input and namespace filters",
+				goldenFileName: "otlp-namespace-filters.yaml",
+				pipelines: []telemetryv1alpha1.MetricPipeline{
+					testutils.NewMetricPipelineBuilder().
+						WithName("cls").
+						WithOTLPInput(true, testutils.IncludeNamespaces("apps", "services")).
+						WithOTLPOutput(testutils.OTLPEndpoint("https://localhost")).Build(),
+				},
+			},
+			{
+				name:           "pipeline with multiple input types and mixed configurations",
+				goldenFileName: "multiple-inputs-mixed.yaml",
+				pipelines: []telemetryv1alpha1.MetricPipeline{
+					testutils.NewMetricPipelineBuilder().
+						WithName("cls").
+						WithRuntimeInput(true, testutils.IncludeNamespaces("default")).
+						WithPrometheusInput(true, testutils.ExcludeNamespaces("kube-system")).
+						WithIstioInput(false).
+						WithOTLPInput(true).
+						WithOTLPOutput(testutils.OTLPEndpoint("https://localhost")).Build(),
+				},
+			},
+			{
+				name:           "pipeline with runtime input specific resources enabled",
```

Comment:
We used to have a unit test for each filter processor for runtime resource metric
Can you add these tests?
I mean the equivalent of these tests https://github.com/kyma-project/telemetry-manager/pull/2438/files#diff-760faa168ccdeb0e75db0b9974c10d38bc529c6ffdb7c7bd3a2d5c5375c97d68L298-L476